sharding/nodesCoordinator/indexHashedNodesCoordinatorLite.go

```diff
@@ -0,0 +1,807 @@
+			}
+		}
+	}
+	return false, 0
+}
+
+func (ihgs *IndexHashedNodesCoordinatorLite) createPublicKeyToValidatorMap(
+	eligible map[uint32][]Validator,
+	waiting map[uint32][]Validator,
+) map[string]*ValidatorWithShardID {
+	publicKeyToValidatorMap := make(map[string]*ValidatorWithShardID)
+	for shardId, shardEligible := range eligible {
+		for i := 0; i < len(shardEligible); i++ {
+			publicKeyToValidatorMap[string(shardEligible[i].PubKey())] = &ValidatorWithShardID{
+				Validator: shardEligible[i],
+				ShardID:   shardId,
+			}
+		}
+	}
+	for shardId, shardWaiting := range waiting {
+		for i := 0; i < len(shardWaiting); i++ {
+			publicKeyToValidatorMap[string(shardWaiting[i].PubKey())] = &ValidatorWithShardID{
+				Validator: shardWaiting[i],
+				ShardID:   shardId,
+			}
+		}
+	}
+
+	return publicKeyToValidatorMap
+}
+
+func (ihgs *IndexHashedNodesCoordinatorLite) computeShardForSelfPublicKey(nodesConfig *EpochNodesConfig) (uint32, bool) {
+	pubKey := ihgs.selfPubKey
+	selfShard := ihgs.shardIDAsObserver
+	epNodesConfig, ok := ihgs.nodesConfig[ihgs.currentEpoch]
+	if ok {
+		log.Trace("computeShardForSelfPublicKey found existing config",
+			"shard", epNodesConfig.ShardID,
+		)
+		selfShard = epNodesConfig.ShardID
+	}
+
+	found, shardId := searchInMap(nodesConfig.EligibleMap, pubKey)
+	if found {
+		log.Trace("computeShardForSelfPublicKey found validator in eligible",
+			"epoch", ihgs.currentEpoch,
+			"shard", shardId,
+			"validator PK", pubKey,
+		)
+		return shardId, true
+	}
+
+	found, shardId = searchInMap(nodesConfig.WaitingMap, pubKey)
+	if found {
+		log.Trace("computeShardForSelfPublicKey found validator in waiting",
+			"epoch", ihgs.currentEpoch,
+			"shard", shardId,
+			"validator PK", pubKey,
+		)
+		return shardId, true
+	}
+
+	found, shardId = searchInMap(nodesConfig.LeavingMap, pubKey)
+	if found {
+		log.Trace("computeShardForSelfPublicKey found validator in leaving",
+			"epoch", ihgs.currentEpoch,
+			"shard", shardId,
+			"validator PK", pubKey,
+		)
+		return shardId, true
+	}
+
+	log.Trace("computeShardForSelfPublicKey returned default",
+		"shard", selfShard,
+	)
+	return selfShard, false
+}
+
+// ConsensusGroupSize returns the consensus group size for a specific shard
+func (ihgs *IndexHashedNodesCoordinatorLite) ConsensusGroupSize(
+	shardID uint32,
+) int {
+	if shardID == core.MetachainShardId {
+		return ihgs.metaConsensusGroupSize
+	}
+
+	return ihgs.shardConsensusGroupSize
+}
+
+// GetLastEpochConfig returns the last epoch from nodes config
+func (ihgs *IndexHashedNodesCoordinatorLite) GetLastEpochConfig() uint32 {
+	ihgs.mutNodesConfig.Lock()
+	defer ihgs.mutNodesConfig.Unlock()
+
+	lastEpoch := uint32(0)
+	for epoch := range ihgs.nodesConfig {
+		if lastEpoch < epoch {
+			lastEpoch = epoch
+		}
+	}
+
+	return lastEpoch
+}
+
+// GetNumTotalEligible returns the number of total eligible accross all shards from current setup
+func (ihgs *IndexHashedNodesCoordinatorLite) GetNumTotalEligible() uint64 {
```

Comment:
Idk what to say about this base implementation which has all of his members private, but has getters and setters for _almost_ all of them. Seems like `java old-school` programming where you define setters and getters for all private members.
Why not move this in the same package with the class that is extending its functionality(`indexHashedNodesCoordinator`)? Then, you can embed it and don't have to define all these getters and setters
 Or was it designed like this on purpose?